packages/node/src/config/index.ts

```diff
@@ -12,9 +17,17 @@ function parseOises(oises: OIS[]): OIS[] {
   });
 }
 
-export function parseConfig(config: any): Config {
-  const ois = parseOises(config.ois);
-  return { ...config, ois };
+export function parseConfig(configPath: string, secrets: Record<string, string | undefined>): Config {
+  const config = fs.readFileSync(configPath, 'utf8');
+  const interpolatedConfig = template(config, {
```

Comment:
The only thing I don't like about interpolating everything all at once and using that config, is that we might accidentally leak credentials somewhere when logging a message that includes the value or in a stack trace if something crashes.

I guess the alternative would be to check every field before it's used but that seems a bit worse.

Reply:
That's the thing, we would have to check EVERY field because the interpolation can be done anywhere in the config :confused:

I guess we could introduce a custom logging that would replace any sensitive data (those replaced from env variables) with `***` or something. But other that that, I'm not sure what else we can do here.

Reply:
Yeah exactly. I'm not sure either. I'm not sure how we would determine which variables to replace as `***` since it's already interpolated at that point. I was just pointing it out as it was something we might want to consider.

I think in almost all cases, someone who has access to the logs will have access to the environment secrets. It's the edge cases where they don't have access to the secrets that I was considering.

Reply:
Is it even a valid edge case? I can't think of such example